metrics/timers.go

```diff
@@ -0,0 +1,87 @@
+package metrics
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"gx/ipfs/QmNVpHFt7QmabuVQyguf8AbkLDZoFh7ifBYztqijYT1Sd2/go.opencensus.io/stats"
+	"gx/ipfs/QmNVpHFt7QmabuVQyguf8AbkLDZoFh7ifBYztqijYT1Sd2/go.opencensus.io/stats/view"
+)
+
+// Registry holds a list of measurements
+type Registry struct {
+	msurMu   sync.Mutex
+	measures map[string]*Float64Timer
+}
+
+// NewRegistry creates a new registry
+func NewRegistry() *Registry {
+	return &Registry{
+		measures: make(map[string]*Float64Timer),
+	}
+}
+
+// NewTimer creates a Float64Timer that wraps an opencensus flat64 measurement.
+// The time defaults to milliseconds.
+func (r *Registry) NewTimer(name, desc string) *Float64Timer {
+	r.msurMu.Lock()
+	defer r.msurMu.Unlock()
+	timer, ok := r.measures[name]
+	if ok {
+		return timer
+	}
```

Comment:
Consider failing hard here - it's probably an inadvertent error and two semantically different measurements will be conflated.

Reply:
This comment has made me realize the registry isn't needed, removing it.